modules/evm-accounts/src/lib.rs

```diff
@@ -236,14 +232,43 @@ impl<T: Config> Pallet<T> {
 
 	#[cfg(any(feature = "runtime-benchmarks", feature = "std"))]
 	// Constructs a message and signs it.
-	pub fn eth_sign(secret: &libsecp256k1::SecretKey, what: &[u8], extra: &[u8]) -> EcdsaSignature {
+	pub fn eth_sign(secret: &libsecp256k1::SecretKey, what: &[u8], extra: &[u8]) -> Eip712Signature {
 		let msg = keccak_256(&Self::ethereum_signable_message(&to_ascii_hex(what)[..], extra));
 		let (sig, recovery_id) = libsecp256k1::sign(&libsecp256k1::Message::parse(&msg), secret);
 		let mut r = [0u8; 65];
 		r[0..64].copy_from_slice(&sig.serialize()[..]);
 		r[64] = recovery_id.serialize();
-		EcdsaSignature::from_slice(&r)
+		r
 	}
+
+	fn verify_eip712_signature(eth_address: EvmAddress, sig: &[u8; 65]) -> Option<H160> {
+		let domain_hash = keccak256!("EIP712Domain(string name,string version,uint256 ss58Id,bytes32 salt)");
+		let tx_type_hash = keccak256!("Transaction(address claim)");
```

Comment:
```suggestion
		let tx_type_hash = keccak256!("Transaction(string address)");
```
The message payload should be the Substrate account in hex format.

Reply:
the substrate address and the evm address being claimed right?

Reply:
we don't need evm address in the signing payload because that's recovered from signature. it is already included from the claim extrinsic parameter so we just compare it to the recovered one to make sure it matches